fe/fe-core/src/main/java/com/starrocks/journal/bdbje/BDBJournalCursor.java

```diff
@@ -79,54 +80,112 @@ private BDBJournalCursor(BDBEnvironment env, long fromKey, long toKey) throws Ex
                 break;
             }
         }
+        nextDbPositionIndex -= 1;
```

Comment:
In the old implementation, databases are opened in the constructor method as well as the `next()` method. This brings some side effects.

1. In the `next()` method, we have to open a new database only after a failed attempt to read the current database, which leads to some very confusing if-else statements.
2. The constructor would have to deal with the BDB environment, which is against intuition.

As a result, I now shrink the operation of opening a database only in the `next()` function. 

You may think that the name of `nextDbPositionIndex` is confusing in the constructor's scope, but I think the value is reasonable since it's only an initialized value, and it indeed represents the next index of database that we are going to open in the `next()` method.

Reply:
Cool.